### PR TITLE
[FLINK-31735][docs] Document 'plan' field as object

### PR DESCRIPTION
**Pull request by zentol**
Fixes an issue where `plan` field of the JobDetailsInfo was documented as a string, when in fact we write the json plan as a raw object.

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobDetailsInfo.java [hunk 1]
```diff
@@ -110,8 +113,8 @@ public class JobDetailsInfo implements ResponseBody {
     private final Map<ExecutionState, Integer> jobVerticesPerState;
 
     @JsonProperty(FIELD_NAME_JSON_PLAN)
-    @JsonRawValue
-    private final String jsonPlan;
+    @JsonSerialize(using = RawObjectSerializer.class)
+    private final Object jsonPlan;
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobDetailsInfo.java:117]: Double-checking if we can re-use JobPlanInfo#RawJson

File: docs/static/generated/rest_v1_dispatcher.yml [hunk 2]
```diff
@@ -2304,7 +2304,7 @@ components:
           type: integer
           format: int64
         plan:
-          type: string
```
Comment (dmvk) [inline docs/static/generated/rest_v1_dispatcher.yml:2307]: 🤔 do we also want to have "any" here instead?

Comment (zentol) [reply docs/static/generated/rest_v1_dispatcher.yml:2307]: `any` isn't valie in the openapi spec

**Review by dmvk — COMMENTED** [review]
in the open api spec, we have

```
    RawJson:
      type: object
```

is that actually correct? If I understand it correctly we should be able to pass anything in the rawjson, for example "string"

File: docs/static/generated/rest_v1_dispatcher.yml (hunk 2 above)
Comment (dmvk) [reply docs/static/generated/rest_v1_dispatcher.yml:2307]: ok, what causes confusion to me is that we document it as any in the htlm docs; is there any way we can document is as 'object' there?

**Comment by dmvk** [comment]
nvm, it's correct, it just doesn't align with the HTML docs

File: docs/static/generated/rest_v1_dispatcher.yml (hunk 2 above)
Comment (zentol) [reply docs/static/generated/rest_v1_dispatcher.yml:2307]: > is there any way we can document is as 'object' there?

Yes; I've pushed a change for this.

**Comment by zentol** [comment]
> If I understand it correctly we should be able to pass anything in the rawjson, for example "string"

Technically you are correct, but we only use RawJson for objects. Should this change the docs will be wrong again.

**Review by dmvk — APPROVED** [review]
LGTM 👍 Great stuff!

**Pull request merged**